--- a/art/libdexfile/dex/dex_file.cc	2023-09-20 08:30:55.735009084 +0800
+++ b/art/libdexfile/dex/dex_file.cc	2023-09-20 08:18:40.434577208 +0800
@@ -600,6 +600,31 @@
   return dex::ProtoIndex(it.GetJavaValue().i);
 }
 
+std::ostream& operator<<(std::ostream& os, EncodedArrayValueIterator::ValueType rhs) {
+  switch (rhs) {
+    case EncodedArrayValueIterator::kByte: os << "Byte"; break;
+    case EncodedArrayValueIterator::kShort: os << "Short"; break;
+    case EncodedArrayValueIterator::kChar: os << "Char"; break;
+    case EncodedArrayValueIterator::kInt: os << "Int"; break;
+    case EncodedArrayValueIterator::kLong: os << "Long"; break;
+    case EncodedArrayValueIterator::kFloat: os << "Float"; break;
+    case EncodedArrayValueIterator::kDouble: os << "Double"; break;
+    case EncodedArrayValueIterator::kMethodType: os << "MethodType"; break;
+    case EncodedArrayValueIterator::kMethodHandle: os << "MethodHandle"; break;
+    case EncodedArrayValueIterator::kString: os << "String"; break;
+    case EncodedArrayValueIterator::kType: os << "Type"; break;
+    case EncodedArrayValueIterator::kField: os << "Field"; break;
+    case EncodedArrayValueIterator::kMethod: os << "Method"; break;
+    case EncodedArrayValueIterator::kEnum: os << "Enum"; break;
+    case EncodedArrayValueIterator::kArray: os << "Array"; break;
+    case EncodedArrayValueIterator::kAnnotation: os << "Annotation"; break;
+    case EncodedArrayValueIterator::kNull: os << "Null"; break;
+    case EncodedArrayValueIterator::kBoolean: os << "Boolean"; break;
+    default: os << "EncodedArrayValueIterator::ValueType[" << static_cast<int>(rhs) << "]"; break;
+  }
+  return os;
+}
+
 // Checks that visibility is as expected. Includes special behavior for M and
 // before to allow runtime and build visibility when expecting runtime.
 std::ostream& operator<<(std::ostream& os, const DexFile& dex_file) {
