diff --git a/src/protobuf/CMakeLists.txt b/src/protobuf/CMakeLists.txt
index 08cb330..85882a2 100644
--- a/src/protobuf/CMakeLists.txt 2023-09-20 13:56:08.461395523 +0800
+++ b/src/protobuf/CMakeLists.txt 2023-09-20 13:57:08.561395525 +0800
@@ -298,7 +298,8 @@ endif (MSVC)
 include_directories(
   ${ZLIB_INCLUDE_DIRECTORIES}
   ${protobuf_BINARY_DIR}
-  ${protobuf_SOURCE_DIR}/src)
+  ${protobuf_SOURCE_DIR}/src
+  ${protobuf_SOURCE_DIR}/config)
 
 if (protobuf_UNICODE)
   add_definitions(-DUNICODE -D_UNICODE)
diff --git a/src/protobuf/cmake/libprotobuf-lite.cmake b/src/protobuf/cmake/libprotobuf-lite.cmake
index 08cb330..85882a2 100644
--- a/src/protobuf/cmake/libprotobuf-lite.cmake 2023-09-20 13:56:08.461395523 +0800
+++ b/src/protobuf/cmake/libprotobuf-lite.cmake 2023-09-20 13:57:08.561395525 +0800
@@ -102,7 +102,7 @@ if(protobuf_LINK_LIBATOMIC)
   target_link_libraries(libprotobuf-lite PRIVATE atomic)
 endif()
 if(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
-  target_link_libraries(libprotobuf-lite PRIVATE log)
+  target_include_directories(libprotobuf-lite PUBLIC ${protobuf_SOURCE_DIR}/android)
 endif()
 target_include_directories(libprotobuf-lite PUBLIC ${protobuf_SOURCE_DIR}/src)
 if(protobuf_BUILD_SHARED_LIBS)
diff --git a/src/protobuf/cmake/libprotobuf.cmake b/src/protobuf/cmake/libprotobuf.cmake
index 08cb330..85882a2 100644
--- a/src/protobuf/cmake/libprotobuf.cmake 2023-09-20 13:56:08.461395523 +0800
+++ b/src/protobuf/cmake/libprotobuf.cmake 2023-09-20 13:57:08.561395525 +0800
@@ -118,7 +118,7 @@ if(protobuf_LINK_LIBATOMIC)
   target_link_libraries(libprotobuf PRIVATE atomic)
 endif()
 if(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
-  target_link_libraries(libprotobuf PRIVATE log)
+  target_include_directories(libprotobuf PRIVATE ${protobuf_SOURCE_DIR}/android)
 endif()
 target_include_directories(libprotobuf PUBLIC ${protobuf_SOURCE_DIR}/src)
 if(protobuf_BUILD_SHARED_LIBS)